src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+const lis = tree.querySelectorAll('li');
+
+lis.forEach((el) => {
+  const text = el.childNodes[0].nodeValue.trim();
+  const span = document.createElement('span');
+
+  span.textContent = text;
+  el.childNodes[0].replaceWith(span);
+});
+
+tree.addEventListener('click', (ev) => {
+  ev.stopPropagation();
+
+  const span = ev.target.closest('span');
```

Comment:
The use of `closest('span')` is correct for finding the nearest ancestor span, but ensure that the event target is indeed a descendant of a span. Otherwise, this could return `null`.